### ISSUE TITLE
Support unix socket connection to MySQL and PostgreSQL

### ISSUE THREAD

**Issue by darinpope** [shlinkio/shlink#833]
#### Summary

If you use Google Cloud Run along with Google Cloud SQL, the only option available uses unix sockets when using the currently GA Public IP offerings. Extend the connection configuration to support unix sockets.

#### Documentation

- https://cloud.google.com/sql/docs/mysql/connect-run#php
- https://cloud.google.com/sql/docs/postgres/connect-run#php

#### Note

- Although you can use the standard connection methods with a Private IP Cloud SQL offering, private IP is still in beta for both Cloud Run and Cloud SQL. By providing unix sockets, that allows us to use the GA offerings.

**Comment by acelaya** [comment]
Sounds like something nice to have.

**Comment by acelaya** [comment]
Hey @darinpope. The change is almost ready, but I'm having some dificulties to test it.

If I publish a v2.5.0-alpha.1 including this, would you be open to test it on your Google Cloud set-up and confirm if it works?

**Comment by darinpope** [comment]
sure, I can try it out. However, it may be a little later in the week before I can work on it.

**Comment by acelaya** [comment]
Sure, that's perfect. I'll try to publish the alpha version during the week and afterwards, you can test it whenever you can/want.

**Comment by acelaya** [comment]
So basically, this alpha version (and the next stable version if everything goes right) will allow you to provide the `DB_UNIX_SOCKET` env var to the docker image, in order to define the socket for MySQL, MariaDB or Postgres.

It will also support it during the manual installation if you are not using the docker image, but since you mentioned Google Cloud Run, I guess it's the former for you.

**Comment by acelaya** [comment]
Once this is finished, a docker image for v2.5.0-alpha.1 will be available: https://github.com/shlinkio/shlink/runs/1470673278

**Comment by darinpope** [comment]
Upon initial testing, it doesn't appear to work. 

I've set DB_DRIVER to "mysql", as well as my DB_USER and DB_PASSWORD. 

I then set DB_UNIX_SOCKET to "/cloudsql/<connection name>" Reviewing the logs, I'm not sure if unix_socket is being picked up correctly or not. 

The logs are showing:

```
2020-11-30 21:18:09.153 EST
Creating fresh database if needed...
Default
2020-11-30 21:18:15.027 EST
[2020-12-01T02:18:14.966155+00:00] [NULL] Shlink.INFO - Successfully acquired the "db:create" lock.
Default
2020-11-30 21:18:15.027 EST
[2020-12-01T02:18:15.027486+00:00] [NULL] Shlink.INFO - Expiration defined for "db:create" lock for "90" seconds.
Default
2020-11-30 21:18:15.136 EST
Default
2020-11-30 21:18:15.137 EST
In AbstractMySQLDriver.php line 93:
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or dir
Default
2020-11-30 21:18:15.137 EST
ectory
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
In PDOConnection.php line 31:
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
SQLSTATE[HY000] [2002] No such file or directory
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
In PDOConnection.php line 27:
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
SQLSTATE[HY000] [2002] No such file or directory
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
Default
2020-11-30 21:18:15.137 EST
db:create
Default
2020-11-30 21:18:15.137 EST
Warning
2020-11-30 21:18:15.519 EST
Container called exit(1).
```
Is there some debugging/logging I can set as an environment variable to get more info?

**Comment by darinpope** [comment]
After a false start, it appears to install and run, but I need to do more validation work in GCP. More to come...
